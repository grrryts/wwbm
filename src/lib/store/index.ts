import rootReducer from './root.reducer';
import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
  reducer: rootReducer,
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export default store;
