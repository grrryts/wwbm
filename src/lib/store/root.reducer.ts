import coreSlice from './slices/core.slice';
import gameReducer from './slices/game.slice';
import { combineReducers } from '@reduxjs/toolkit';

const rootReducer = combineReducers({
  core: coreSlice,
  game: gameReducer,
});

export default rootReducer;
