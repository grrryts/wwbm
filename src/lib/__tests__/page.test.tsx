import { Provider } from 'react-redux';

import store from '@/lib/store';
import HomeView from '@/ui/views/home-view';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

it('renders the HomeView and matches snapshot', () => {
  const container = render(
    <Provider store={store}>
      <HomeView />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
