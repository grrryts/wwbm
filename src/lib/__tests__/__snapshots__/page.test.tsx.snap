// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the HomeView and matches s 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-9ay3d9"
      >
        <div
          class="MuiBox-root css-k7yw13"
        >
          <div
            class="MuiBox-root css-1mhkxiw"
          >
            <div
              class="MuiBox-root css-1dvwhmv"
            >
              <img
                alt="thumbs-up"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
              />
            </div>
            <div
              class="MuiBox-root css-1uiamur"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
              >
                Who wants to be a millionaire?
              </h1>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-9ay3d9"
    >
      <div
        class="MuiBox-root css-k7yw13"
      >
        <div
          class="MuiBox-root css-1mhkxiw"
        >
          <div
            class="MuiBox-root css-1dvwhmv"
          >
            <img
              alt="thumbs-up"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
            />
          </div>
          <div
            class="MuiBox-root css-1uiamur"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
            >
              Who wants to be a millionaire?
            </h1>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Start
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the HomeView and matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-9ay3d9"
      >
        <div
          class="MuiBox-root css-k7yw13"
        >
          <div
            class="MuiBox-root css-1mhkxiw"
          >
            <div
              class="MuiBox-root css-1dvwhmv"
            >
              <img
                alt="thumbs-up"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
              />
            </div>
            <div
              class="MuiBox-root css-1uiamur"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
              >
                Who wants to be a millionaire?
              </h1>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-9ay3d9"
    >
      <div
        class="MuiBox-root css-k7yw13"
      >
        <div
          class="MuiBox-root css-1mhkxiw"
        >
          <div
            class="MuiBox-root css-1dvwhmv"
          >
            <img
              alt="thumbs-up"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
            />
          </div>
          <div
            class="MuiBox-root css-1uiamur"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
            >
              Who wants to be a millionaire?
            </h1>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Start
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-9ay3d9"
      >
        <div
          class="MuiBox-root css-k7yw13"
        >
          <div
            class="MuiBox-root css-1mhkxiw"
          >
            <div
              class="MuiBox-root css-1dvwhmv"
            >
              <img
                alt="thumbs-up"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
              />
            </div>
            <div
              class="MuiBox-root css-1uiamur"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
              >
                Who wants to be a millionaire?
              </h1>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-9ay3d9"
    >
      <div
        class="MuiBox-root css-k7yw13"
      >
        <div
          class="MuiBox-root css-1mhkxiw"
        >
          <div
            class="MuiBox-root css-1dvwhmv"
          >
            <img
              alt="thumbs-up"
              data-nimg="fill"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhand_thumbs_up.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; object-position: center; color: transparent;"
            />
          </div>
          <div
            class="MuiBox-root css-1uiamur"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-10l7o3s-MuiTypography-root"
            >
              Who wants to be a millionaire?
            </h1>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Start
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
